action.yml

```
@@ -336,26 +344,60 @@ runs:
+              head: `${context.payload.workflow_run.head_repository.owner.login}:${context.payload.workflow_run.head_branch}`,
+              state: 'open'
+            });
+            
+            if (pulls.data.length > 0) {
+              // Try to find the PR whose head SHA matches the workflow run's head_sha
+              const matchingPr = pulls.data.find(pr => pr.head.sha === context.payload.workflow_run.head_sha);
+              
+              if (matchingPr) {
+                core.setOutput('number', matchingPr.number);
+                core.setOutput('sha', matchingPr.head.sha);
+              } else {
+                // Fallback: use the first PR, but warn in the summary
+                core.setOutput('number', pulls.data[0].number);
+                core.setOutput('sha', context.payload.workflow_run.head_sha);
+                core.warning(`Multiple PRs found for branch ${context.payload.workflow_run.head_branch}, but none matched head_sha. Using the first PR (#${pulls.data[0].number}).`);
+              }
+            } else {
+              core.warning(`No open PRs found for branch ${context.payload.workflow_run.head_branch} from ${context.payload.workflow_run.head_repository.owner.login}`);
+            }
+          }
+
     - name: Find Comment to update
-      if: ${{ inputs.set-pr-comment == 'true' && (github.event_name == 'pull_request' || github.event_name == 'pull_request_target') }}
+      if: ${{ inputs.set-pr-comment == 'true' && (github.event_name == 'pull_request' || github.event_name == 'pull_request_target' || (github.event_name == 'workflow_run' && steps.pr-number.outputs.number)) }}
       uses: peter-evans/find-comment@v3
       id: fc
       with:
-        issue-number: ${{ github.event.pull_request.number }}
+        issue-number: ${{ github.event.pull_request.number || steps.pr-number.outputs.number }}
         comment-author: 'github-actions[bot]'
         body-includes: '🚀 Build'
         token: ${{ inputs.github-token }}
 
     - name: Create or update comment
-      if: ${{ inputs.set-pr-comment == 'true' && (github.event_name == 'pull_request' || github.event_name == 'pull_request_target') }}
+      if: ${{ inputs.set-pr-comment == 'true' && (github.event_name == 'pull_request' || github.event_name == 'pull_request_target' || (github.event_name == 'workflow_run' && steps.pr-number.outputs.number)) }}
       uses: peter-evans/create-or-update-comment@v4
       with:
         token: ${{ inputs.github-token }}
         comment-id: ${{ steps.fc.outputs.comment-id }}
-        issue-number: ${{ github.event.pull_request.number }}
+        issue-number: ${{ github.event.pull_request.number || steps.pr-number.outputs.number }}
         body: |
           ### 🚀 Build Preview on IPFS ready
-          - 🔎 Commit: ${{ github.event.pull_request.head.sha || github.sha }}
+          - 🔎 Commit: ${{ github.event.pull_request.head.sha || steps.pr-number.outputs.sha || github.sha }}
```

```suggestion
          - 🔎 Commit: ${{ github.event.pull_request.head.sha || steps.pr-number.outputs.sha }}
```

I'm pretty sure we never want to use `github.sha` for pull requests. Because that will be the head of the branch we're merging into.